Synchronized GitHub repo labels

jromero opened the pull request:
[readable](https://github.com/jromero/rfcs/blob/gh-repo-labels/text/0000-gh-repo-labels.md)

nebhale commented on text/0000-gh-repo-labels.md:
````diff
@@ -0,0 +1,104 @@
+        color = ""        # an RGB hex color value 
+        description = ""  # the description of the label 
+    ```
+- A GitHub action that keeps the labels in sync.
````
Concur with removing the implementation detail.

jromero replied:
I don't have a problem removing it. I do struggle to find the balance between how much information is too little (vague) or too much (implementation details). My understanding is that RFCs can be as close to specific as necessary and are more as a method for guidance and conversation and not to be binding to any specification. That's where specs come in. :D

jromero replied:
Made it a little more ambiguous.

ekcasey commented on text/0000-gh-repo-labels.md:
```diff
@@ -0,0 +1,104 @@
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Where does this live? [community](https://github.com/buildpacks/community) or new "plumbing" repo?
```
I would vote community to contain the sprawl

zmackie commented:
I have a couple general comments:
- This would be super useful, espcially for cross repo tags for issue and PR status + roadmap eg. `roadmap/ci-ci` . This makes searching across org by label super nice.
- It might be ok to start with manual and automate from there, but maybe a first step is actually agreeing on the labelling schema and a handful of labels we want. That seems seperate from the syncing implementation.

jromero commented on text/0000-gh-repo-labels.md:
````diff
@@ -0,0 +1,104 @@
+```
+
+A GitHub actions will be triggered when changes to this file are made. The action will ensure the `global` labels exist
+on all repositories in the org. It would also ensure that any labels specific to a repository (`repo.<repo-name>`) exist
````
This is really good feedback. I changed the format so that it's easier to select (include/exclude) repositories.

jromero commented:
@simonjjones 

> I think the drawbacks described here are accurate and do present a substantial challenge. I'd really like to understand what the burden of labels potentially becoming inconsistent across our repos is.

I think this is sufficiently detailed in the motivation. If it isn't please let me know what sort of information you are looking for.

> How many of them have enough activity, or a similar enough workflow to make these labels meaningful?

I don't think the current activity should be a measure for the need here. The intent is to provide consistency to improve the use of labels and the benefits brought by them. In my case, and as @ekcasey mentioned, automation is a goal here. Additionally, discoverability, as @zmackie mentioned is another.

> Should the synchronisation happen to archived repos?

No. Added to RFC.

>Is there any way to prevent labels being added manually, that would later be cleaned up by the automation?

There is a way. See `Unresolved Questions`. 

> Could this lead to a negative experience, especially for unfamiliar contributors?

Added to drawbacks.

> How frequently should synchronisation happen? If we only synchronise when changes to labels.toml are detected, we'd potentially have some long lived manually created labels that are then suddenly deleted.

If we prevent manual labels from existing (see above), then this is a non-issue right?

jromero commented:
@ekcasey 

> However, we could start applying consistent labels and introducing automation that leverages those labels without first automating label creation or achieving perfect consistency with colors/descriptions. I would prefer that we manually tinker to find the set of labels we want before we lock it in stone and gate changes with the PR approval process.

@zmackie 

> It might be ok to start with manual and automate from there, but maybe a first step is actually agreeing on the labelling schema and a handful of labels we want. That seems seperate from the syncing implementation.

The platform team is currently in this phase. We are still really early in the discovery phase but chose to put up this RFC concurrently (and so that I would defer it indefinitely).

To that effect, we are currently manually trying to determine what the labels should be with @zmackie's input. We should have an RFC for **issue workflow** very soon that will point out proposed labels. This RFC is solely for the proposal of syncing them.

simonjjones commented:
Held - awaiting definition of the specific labels we want to include - @jromero to follow up

thephw commented:
@jromero do y'all have some prior art you're looking to leverage? I've got a similar bit of code from the consultancy that I could rework a bit. Basic interface was this: 
<img width="780" alt="Screen Shot 2020-03-11 at 3 37 23 PM" src="https://user-images.githubusercontent.com/1013461/76457063-76fd6a00-63ae-11ea-887e-527b6236def0.png">

jromero commented:
The specific labels that would be synchronized added to https://github.com/buildpacks/rfcs/pull/58.

nebhale commented:
I move for an FCP, ending 29 April, for merge.
